lib/buffer.js

```diff
@@ -933,8 +940,12 @@ Buffer.prototype.readUInt32BE = function(offset, noAssert) {
 
 
 Buffer.prototype.readIntLE = function(offset, byteLength, noAssert) {
+  if (!noAssert)
+    checkByteLength(byteLength);
```

Comment:
I agree that ideally we would combine the two, but there could easily be breakage if the current (prior to this PR) assertions are moved, so that would definitely be a semver-major change *if* we wanted to make that change at all. Thoughts @nodejs/collaborators?